### PR TITLE
fix(python/trezorctl): firmware download does not require connected trezor

### PR DESCRIPTION
**Pull request by grdddj**
Connected with issue #1715
- `trezorctl firmware download` command no longer requires a trezor device to be connected

### CODE REVIEW

**Comment by grdddj** [comment]
Fixup commit 23e2422:
- forgotten `bitcoin_only` argument in function getting release details

**Comment by grdddj** [comment]
Fixup commit 2b6c5c8:
- removing the optionality from `version` argument from new function - it is mandatory there

File: python/src/trezorlib/cli/firmware.py [hunk 1]
```diff
@@ -183,6 +183,46 @@ def get_all_firmware_releases(
     return releases
 
 
+def get_url_and_fingerprint_from_release(
+    release: dict,
+    bitcoin_only: bool = False,
```
Comment (matejcik) [inline python/src/trezorlib/cli/firmware.py:188]: this should not be optional either

File: python/src/trezorlib/cli/firmware.py [hunk 2]
```diff
@@ -183,6 +183,46 @@ def get_all_firmware_releases(
...
+        click.echo(f"Unsupported URL found: {url}")
+        sys.exit(1)
+    final_url = "https://data.trezor.io/" + url[len(url_prefix) :]
+
+    return final_url, fingerprint
+
+
+def find_specified_firmware_version(
+    version: str,
+    beta: bool = False,
```
Comment (matejcik) [inline python/src/trezorlib/cli/firmware.py:209]: ISTM these shouldn't be optional either, nor the arguments of `find_best_firmware_version`

are there any calls to these functions that omit these arguments?

**Comment by grdddj** [comment]
Fixup commit 1c6db49:
- removing the optionality of some function arguments
- changing `exit(1)` to `sys.exit(1)`, which is being used elsewhere

**Comment by matejcik** [comment]
LGTM. please squash & rebase. if you now have the rights, please push the branch directly to firmware repo so that we get a CI run

**Comment by grdddj** [comment]
I tried to push into this repo directly, but still getting permission issues:
image

**Comment by matejcik** [comment]
i pushed it for you, waiting for CI to go green

**Comment by grdddj** [comment]
Is it some different (more thorough) CI than we have here in this PR?

image

**Comment by matejcik** [comment]
the gitlab part doesn't run for external PRs, the branch must be local

**Comment by grdddj** [comment]
Seems like the changelog is missing, should I include it?

**Comment by matejcik** [comment]
no, this is not "changed between releases" and adding the feature already has a changelog entry. let's ignore that failure

otherwise green so merging, thanks!

**Comment by grdddj** [comment]
Nice, thank you for cooperation!

**Pull request merged**